### DISCUSSION TITLE
want to compile server for linux to make a standalone image for it like soundgrid.

### DISCUSSION THREAD

**Discussion by DCTech-lab** [apohl79/audiogridder#487] [Ideas]
Hi,
I want to compile server for linux to make a stand alone image for it like soundgrid server. Could you please help me.

Best Regards

**Comment by DCTech-lab** [comment]
ok this is related to  "(Request) Linux Server plans?! #214"

But little extra I need for standalone server that GUI part need to be separated from code. At present it has scanner\controller and main server code in same application. Idea is to separate the gui controller (as in multirack control panel) from server (the spawned child process part). Then server can run as service in windows or as demon on linux responding to the calls of controller (surface in soundgrid terminology) 
Once server on linux compiles, i can start working on soundgrid like protocol for daw n server discovery.

**Comment by apohl79** [comment]
The server code is not supported on linux right now. Mainly the screen capturing code and keyboard/mouse support has to be ported. You might be able to get it compiled and get the audio processing working without it.

As you did mention soundgrid a few times, I don't understand what your goal is. As far as I know, soundgrids network I/O is based on a custom layer 2 protocol, while AG uses standard TCP right now. Maybe you can explain that a bit more. Also, I'm pretty sure soundgrid does not rely on screen capturing for the UI part. So overall, I don't understand where you want to go, and how AG does fit here.

**Comment by DCTech-lab** [comment]
Thanks for answering. My aim is to make an application like emotionLV1 (live mixer) product of waves. I encountered this project and in theory it looks similar to that. That’s why I wanted to enhance it like that.  Yes their soundgrid protocol is layer over PPP. I wanted to implement similar to reduce upper layer processing overhead and delay and clock synchronisation. But I encountered a test Audiogridder  vs SGServer  (bottom post by javila77 on page DIY Waves Soundgrid Server - 2020 Edition - Page 3 - Gearspace (gearslutz.com)) and looks  like I m jumping too fast. What I need is to work on mixer part first. Clock synchronisation is feature  for later stages when a basic product is ready. I will explore some open source projects for that. Thanks,  Sent from Mail for Windows 10 From: Andreas PohlSent: 10 April 2021 23:32To: apohl79/audiogridderCc: DCTech-lab; AuthorSubject: Re: [apohl79/audiogridder] want to compile server for linux to make a standalone image for it like soundgrid. (#487) The server code is not supported on linux right now. Mainly the screen capturing code and keyboard/mouse support has to be ported. You might be able to get it compiled and get the audio processing working without it.As you did mention soundgrid a few times, I don't understand what your goal is. As far as I know, soundgrids network I/O is based on a custom layer 2 protocol, while AG uses standard TCP right now. Maybe you can explain that a bit more. Also, I'm pretty sure soundgrid does not rely on screen capturing for the UI part. So overall, I don't understand where you want to go, and how AG does fit here.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Reply by jaacMusic** [reply]
Hello DCTech-lab, I was the one who posted on Gearspace as javila77. I understand what you trying to accomplish as I was trying Waves Soundgrid server before I encountered this project. I love the concept of Soundgrid but sadly it only supports Waves' plugins.
@apohl79, I think what he wants is to be able to run the server in a light operating system (Soundgrid server uses a light linux distro) that doesn't require a full blown OS installation that sometimes you have to deal with in regards to security, patches, etc.
Soundgrid server image can be booted from a USB and it is very practical from that perspective.
